modules/widgets/wordpress-post-widget.php

```diff
@@ -27,138 +73,600 @@ public function __construct() {
 				'description' => __( 'Displays a list of recent posts from another WordPress.com or Jetpack-enabled blog.', 'jetpack' ),
 			)
 		);
+
+		/**
+		 * Check the status of the cron task.
+		 */
+		self::check_for_cron();
```

Comment:
I see why you put the cron check here (it's a handy place), but it's a little strange since there can be multiple instances of this class.